File: dotnet/Samples/ContentInsights/EdgeContentInsightsService.cs
using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.ContentInsights.Edge.Request;
using LanguageWeaver.Sdk.ContentInsights.Edge.Result;
using LanguageWeaver.Sdk.Translate.Edge.Request;

namespace LanguageWeaver.Sdk.Samples.ContentInsights;

public class EdgeContentInsightsService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();

        var translateFileRequest = new EdgeTranslateFileRequest
        {
            LanguagePairId = "EngFra_Generic_SRV_TNMV_8_5_x_1",
            InputFormat = Format.Plain,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            },
            // provide full path to the source and output file
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt"),
            OutputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output", "input1-translated.txt")
        };

        var edgeTranslationFileResult = lwClient.TranslateFile(translateFileRequest);
        EdgeContentInsightsRequest edgeContentInsightsRequest =
            new EdgeContentInsightsRequest()
            {
                TranslationId = edgeTranslationFileResult.TranslationId
            };

        EdgeContentInsightsResult edgeContentInsightsResult =
            lwClient.GetContentInsightsForTranslations(edgeContentInsightsRequest);
        // handle result
    }
}

File: java/src/main/java/com/languageweaver/sdk/samples/contentinsights/EdgeContentInsightsService.java
package com.languageweaver.sdk.samples.contentinsights;

import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.common.edge.EdgeLanguageWeaverClient;
import com.languageweaver.sdk.contentinsights.edge.request.EdgeContentInsightsRequest;
import com.languageweaver.sdk.contentinsights.edge.result.EdgeContentInsightsResult;
import com.languageweaver.sdk.translate.edge.request.EdgeTranslateFileRequest;

import java.io.File;
import java.nio.file.Paths;

public class EdgeContentInsightsService {

    public static void main(String[] args) throws Exception {
        try (EdgeLanguageWeaverClient lwClient = new EdgeLanguageWeaverClient().build()) {
            EdgeTranslateFileRequest translateFileRequest = new EdgeTranslateFileRequest()
                    .setLanguagePairId("EngFra_Generic_SRV_TNMV_8_5_x_1")
                    // provide full path to the source file
                    .setInputFile(Paths.get("java", "src", "main", "resources", "input", "input1.txt").toFile().getAbsolutePath())
                    .setOutputFile(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath() + File.separator + "input1-translated.txt")
                    .setInputFormat(Format.PLAIN)
                    .addDictionary("DictionaryName1")
                    .addDictionary("DictionaryName2");

            EdgeContentInsightsRequest edgeContentInsightsRequest = new EdgeContentInsightsRequest()
                    .setTranslationId(lwClient.translateFile(translateFileRequest).getTranslationId());

            final EdgeContentInsightsResult edgeContentInsightsResult = lwClient.getContentInsightsForTranslations(edgeContentInsightsRequest);
            // handle result
        }
    }
}
